### PR TITLE
Add ATLAS style callouts

### PR DESCRIPTION
**Pull request by Sporiff**

### CODE REVIEW

**Comment by Sporiff** [comment]
@bzavhorodskyi this isn't perfect, but I've got the kitchen sink page to show things as I want it to. Let me know what you think.

File: src/components/Callout.astro [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+declare type BannerKind =
+  | "neutral"
+  | "positive"
+  | "warning"
+  | "negative"
+  | "primary";
+
+export interface Props {
+  type: string | undefined;
+  title?: string | undefined;
```
Comment (bzavhorodskyi) [inline src/components/Callout.astro:13]: We can remove " | undefined" when using the "typeName?: string" syntax. 

`type?: string;`
`title?: string`

image

File: src/components/Callout.astro [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+let placeholderTitle = type[0].toUpperCase() + type.slice(1);
+
+if (Astro.slots.has("default")) {
+  description = await Astro.slots.render("default");
+}
+---
+
+<BuildBanner
+  client:only="react"
+  title={title ? title : placeholderTitle}
```
Comment (bzavhorodskyi) [inline src/components/Callout.astro:40]: When we are checking one value for undefined we can use simpler syntax, here are two ways:
1. `title ?? placeholderTitle` - checks only for undefined value, it means if the title value will be falsy(empty string, 0, etc.) this check will be passed
2. `title || placeholderTitle` - checks for all types of falsy values, if the title will be falsy we will use placeholderTitle

**Comment by bzavhorodskyi** [comment]
In general, I`ve checked how it looks and I like it

**Pull request merged**